Add strategy document (#927)
exercism/website-copy: pages/strategy.md

> Exercism is a large complex project with thousands of contributors working across dozens of projects, all trying to achieve ambitious goals. It would quickly become overwhelming if we were all working towards making Exercism perfect, so to keep things manageable, we periodically determine the critical next step for Exercism to achieve its goals and then plan that into a strategy. Our current goal is to make Exercism scalable. We expect our next three goals to be to make Exercism financially sustainable, to make Exercism more accessible to beginners, and to intentionally grow Exercism for the first time.

Reviewer: As this is a long read, I would personally prefer that the goals are separate from this big paragraph:

```suggestion
Exercism is a large complex project with thousands of contributors working across dozens of projects, all trying to achieve ambitious goals. It would quickly become overwhelming if we were all working towards making Exercism perfect, so to keep things manageable, we periodically determine the critical next step for Exercism to achieve its goals and then plan that into a strategy.

Our _current_ goal is to make Exercism scalable.

We expect our next three goals to be:
- to make Exercism financially sustainable,
- to make Exercism more accessible to beginners,
- and to intentionally grow Exercism for the first time.
```
------------

Exercism is a large complex project with thousands of contributors working across dozens of projects, all trying to achieve ambitious goals. It would quickly become overwhelming if we were all working towards making Exercism perfect, so to keep things manageable, we periodically determine the critical next step for Exercism to achieve its goals and then plan that into a strategy.

Our _current_ goal is to make Exercism scalable.

We expect our next three goals to be:
- to make Exercism financially sustainable,
- to make Exercism more accessible to beginners,
- and to intentionally grow Exercism for the first time.

> We present this as an Objective, which will aim to achieve with a Strategy, supported by a series of Tactics. This is commonly known as the OST framework.

Reviewer: ```suggestion
We present this as an **Objective**, which will aim to achieve with a **Strategy**, supported by a series of **Tactics.** This is commonly known as the OST framework.
```

I suggest you highlight these words.

## Objective

**Make Exercism scalable**

> At the time of writing, Exercism has 8,000 new submissions per week. Our mentors currently cannot keep up with the amount of submissions submitted. When they do keep up, the number of submissions almost immediately doubles, as students stick with a Language Track. We are growing by 300-400 new users per day. Many find Exercism too hard to use, but in the future, when that is fixed, we will expect to see the number of submissions double again. We have never actively marketed Exercism. All 300,000 members have come through word of mouth. When we start to market Exercism, we expect to see an increase in usage by two orders of magnitude. Our most important goal at this stage is being able to give our current students a timely mentoring experience, and be able to cope with further growth without overwhelming mentors.

Reviewer: Same as the long read all the way at the start of the document, I would separate the concrete goal from the paragraph, like so (including Katrina's change):

```suggestion
At the time of writing, Exercism has 8,000 new submissions per week. Our mentors currently cannot keep up with the number of submissions submitted. When they do keep up, the number of submissions almost immediately doubles, as students stick with a Language Track. We are growing by 300-400 new users per day. Many find Exercism too hard to use, but in the future, when that is fixed, we will expect to see the number of submissions double again. We have never actively marketed Exercism. All 300,000 members have come through word of mouth. When we start to market Exercism, we expect to see an increase in usage by two orders of magnitude. 

Our most important goal at this stage is being able to give our current students a timely mentoring experience, and be able to cope with further growth without overwhelming mentors.
```

I also don't now if it's preferable to combine so many different subjects into one block. You're talking about scalability, but also that many find exercism too hard which targets the progression lines and learning curve. 

I think it might be good to remove that all together and instead write that "those track that have been undergoing restructure via the project track anatomy" have seen significant increase in the number of submissions, or something similar in your words. That said, since in the **Tactics** part you elaborate on this, I think this paragraph works fine without the entire sentence.

Author: For clarity, I mean "hard to use" as in things like downloading the CLI rather than finding the exercises too hard.

Reviewer: Now that you say it, it makes sense. This is not clear from the wording 🗡 😅

We aim to achieve this goal by the end of 2019.

## Strategy

**Acquire and retain more mentors by making mentoring rewarding and enjoyable**

> Currently, mentoring can be both fulfilling and frustrating. It can be very rewarding to help a student that wants to learn. On the flipside, it can be repetitive, students can be rude, there are few positive feedback loops, and it can feel too much like a job rather than a hobby. In order to retain our existing mentors, and to acquire new ones, we must make mentoring more enjoyable and more rewarding. We must make our mentors feel like that are part of something bigger them them, and that they are contributing to something valuable. A good parallel is wikipedia, where thousands of people contribute (from once a year, to dozens of times per day) because they feel like they are making a tangible difference to something that matters, and because the process itself is straight-forward but (strangely) rewarding.

Reviewer: I think a semicolon separated list reads better here:
```markdown
On the flipside: it can be repetitive, students can be rude; there are few positive feedback loops; and it 
can feel too much like a job rather than a hobby
```

Reviewer: >  We must make our mentors feel like that are part of something bigger them them,

 We must make our mentors feel like that are part of something bigger **than** them

Reviewer: > A good parallel is wikipedia, where

A good parallel is *Wikipedia*, where

Reviewer: >  straight-forward but (strangely) rewarding

 straight-forward **; and** (strangely) rewarding

Author: I'm not a huge fan of semi-colons here - I think commas do the job ok.

## Tactics

### Track Anatomy Project (TAP)

> Some solutions are fun to mentor, some are extremely challenging. Foe example, solutions that require maths lead to mentors trying to tutor someone on algebra, not on coding. Other solutions can be solved in such a variety of ways that the mentor ends up debating the right algorithm, not teaching the idioms of the language.

Reviewer 2: s/Foe/For

Author: ```suggestion
Some solutions are fun to mentor, some are extremely challenging. For example, solutions that require maths lead to mentors trying to tutor someone on algebra, not on coding. Other solutions can be solved in such a variety of ways that the mentor ends up debating the right algorithm, not teaching the idioms of the language.
```

Author: Thanks.

> TAP will fix this by providing a framework to Track Maintainers, which they can work through in order to determine the optimal series of core exercises, and sensible related side exercises. In our prototype Track (Ruby), the result of TAP saw a average reduction in waiting time from one week to half a day, and reduced the queue from hundreds to ~30 at any time. It also saw a doubling in the amount of submissions per week and dramatically increased mentor retention, with the most active weekly number mentors of any track. Encoding this into a framework and applying it to all tracks should have a dramatic affect on mentor enjoyment.

Reviewer 3: > doubling in the amount of submissions

_doubling in the **number** of submissions_

Author: ```suggestion
TAP will fix this by providing a framework to Track Maintainers, which they can work through in order to determine the optimal series of core exercises, and sensible related side exercises. In our prototype Track (Ruby), the result of TAP saw a average reduction in waiting time from one week to half a day, and reduced the queue from hundreds to ~30 at any time. It also saw a doubling in the number of submissions per week and dramatically increased mentor retention, with the most active weekly number mentors of any track. Encoding this into a framework and applying it to all tracks should have a dramatic affect on mentor enjoyment.
```

> The Track Anatomy Project will fix this by providing a framework to Track Maintainers, which they can work through in order to determine the optimal series of core exercises, and sensible related side exercises. In our prototype Track (Ruby), the result of the Track Anatomy Project saw a average reduction in waiting time from one week to half a day, and reduced the queue from hundreds to ~30 at any time. It also saw a doubling in the number of submissions per week and dramatically increased mentor retention, with the most active weekly number mentors of any track. Encoding this into a framework and applying it to all tracks should have a dramatic affect on mentor enjoyment.

Reviewer: > by providing a framework to Track Maintainers

Does one "work through a framework"? Perhaps calling it a tool might better?

> The Track Anatomy Project **fixes** this by providing a **tool** to Track Maintainers, which they can **use** to determine...

Then again, calling it a framework at the end of the paragraph makes complete sense. 

> to all tracks should have a dramatic affect on mentor enjoyment.

Might be good to be explicit that this is positive:

> to all tracks should have a **dramatically positive** affect on mentor enjoyment.

### Automated Mentor Support Project

The majority of submissions are common. It is rare for a student to submit a solution that no-one else has ever submitted. The Track Anatomy Project aims to ensure this is the case, guiding tracks to have core exercises that are usually solved with one of a few common approaches. Automated Mentoring Support aims to utilize this consistency to provide automated approval and/or feedback to students for the majority of submissions to core exercises. This has the following benefits:
- **Reduce student frustration:** We give students a quick feedback loop. If a solution is optimal, let them continue. If it fails in a known way, tell them and let them improve it. This keeps students getting frustrated with waiting for their perfectly-ok solutions to be approved and therefore with the mentors who they perceive to be the source of the slowness.

> - **Dramatically reduce the size of the queue:** Our prototyping has shown that for simple exercises (e.g. TwoFer) we will be able to provide automated mentoring in 90%+ of cases. For more complex exercises, he percentage will be smaller but we expect to be able to provide automated mentoring on 50% of exercises, and on all optimal, or close-to-optimal, solutions.

Reviewer: ```suggestion
- **Dramatically reduce the size of the queue:** Our prototyping has shown that for simple exercises (e.g. TwoFer) we will be able to provide automated mentoring in 90%+ of cases. For more complex exercises, the percentage will be smaller but we expect to be able to provide automated mentoring on 50% of exercises, and on all optimal, or close-to-optimal, solutions.
```

### Provide realistic expectations to students

A key frustration for students is that some Tracks have a response time of hours and others have a response time of weeks. This discrepancy means that we frustrate and lose many students, many of whom could eventually become mentors. We also end up with lots of "abandoned" solutions that we don't know are abandoned until the student doesn't respond to the mentor - something that disheartens mentors and causes them to leave. 

By providing realistic response-time expectations for the students in the UI **before** they put the effort into solving a solution, we can remove this frustration for both students and mentors.

### Remove the barrier to mentoring

> We currently have a deliberate barrier to mentoring with a wall that you have to sign up through. We have put this in place because we want to try and ensure a level of quality control and therefore want to gate-keep and ensure that people who are mentoring feel like they are up to the taks. However, as we add automation, documentation, and a more integrated community, we will be able to remove this barrier and making mentoring more of "something for everyone" on Exercism, rather than something for the elite.

Reviewer 3: > We currently have a deliberate barrier to mentoring with a wall that you have to sign up through.

How about:

_We currently have a deliberate barrier to mentoring, requiring mentors to sign up._

> we want to try and ensure

_try **to** ensure_ I think?

> feel like they are up to the taks

_feel like they are up to the **task**

> rather than something for the elite.

Maybe: _rather than something exclusive._

## Introduction

Exercism is a large complex project with thousands of contributors working across dozens of projects, all trying to achieve ambitious goals. It would quickly become overwhelming if we were all working towards making Exercism perfect, so to keep things manageable, we periodically determine the critical next step for Exercism to achieve its goals and then plan that into a strategy.  We present this using the OST framework - defining an **Objective**, which will aim to achieve with a **Strategy**, implemented by a series of **Tactics**.

Our _current_ objective is to make Exercism scalable. 

We expect our next three objectives to be:
- to make Exercism financially sustainable
- to make Exercism more accessible to beginners
- to intentionally grow Exercism for the first time

## Objective

> **Make Exercism scalable**

Reviewer 4: ```suggestion
###Make Exercism scalable
```
Later in the DOC you you switch to this pattern from `**<say somthing>**`.

Author: @rpottsoh Thanks :) Yeah - this was deliberate. The reason is that there's only one objective and only one strategy, so visually having the two headings look a bit "off" to me. Once it's merged on the website, I'll check again to see if I think that makes sense or whether I should shift to what you suggest.

At the time of writing, Exercism has 8,000 new submissions per week. Our mentors currently cannot keep up with the amount of submissions submitted. When they do keep up, the number of submissions almost immediately doubles, as students stick with a Language Track. We are growing by 300-400 new users per day. Many find Exercism too hard to use, but in the future, when that is fixed, we will expect to see the number of submissions double again. We have never actively marketed Exercism. All 300,000 members have come through word of mouth. When we start to market Exercism, we expect to see an increase in usage by two orders of magnitude. 

Our most important goal at this stage is being able to give our current students a timely mentoring experience, and be able to cope with further growth without overwhelming mentors.

We aim to achieve this goal by the end of 2019.

## Strategy

> **Acquire and retain more mentors by making mentoring rewarding and enjoyable**

Reviewer 4: ```suggestion
###Acquire and retain more mentors by making mentoring rewarding and enjoyable
```
If I have missed the point for the change in formatting mid-stream then disregard.

Some solutions are fun to mentor, some are extremely challenging. For example, solutions that require maths lead to mentors trying to tutor someone on algebra, not on coding. Other solutions can be solved in such a variety of ways that the mentor ends up debating the right algorithm, not teaching the idioms of the language.

We aim to fix this by determining the optimal set and order of core exercises within a track, which will provide a sensible progression for students and be fun to review for mentors.

_Update: We have called this this the [Track Anatomy Project](https://exercism.io/blog/track-anatomy-project). In our prototype Track (Ruby), the result of the Track Anatomy Project saw a average reduction in waiting time from one week to half a day, and reduced the queue from hundreds to ~30 at any time. It also saw a doubling in the number of submissions per week and dramatically increased mentor retention, with the most active weekly number mentors of any track. Encoding this into a framework and applying it to all tracks should have a dramatic positively effect on mentor enjoyment._

### Automatically analyze common solutions

> The majority of submissions are common. It is rare for a student to submit a solution that no-one else has ever submitted. As part of restructing the track, we aim to enforce this as a contraint, guiding tracks to have core exercises that are usually solved with one of a few common approaches. We aim to utilize this consistency to provide automated approval and/or feedback to students for the majority of submissions to core exercises. This has the following benefits:

Reviewer 3: > we aim to enforce this as a constraint

It's not clear what "this" points to.

> The majority of submissions are common. It is rare for a student to submit a solution that no-one else has ever submitted. As part of restructing the track, we aim to enforce this as a constraint, guiding tracks to have core exercises that are usually solved with one of a few common approaches. We aim to utilize this consistency to provide automated approval and/or feedback to students for the majority of submissions to core exercises. This has the following benefits:

Reviewer 3: I think my previous comment got lost in history somewhere (I'm not seeing it, anyway).

> We aim to enforce this as constraint

It's not clear what "this" points to.

> - **Reduce student frustration:** We give students a quick feedback loop. If a solution is optimal, let them continue. If it fails in a known way, tell them and let them improve it. This prevents students from getting frustrated with waiting for their perfectly-ok solutions to be approved and therefore with the mentors who they perceive to be the source of the slowness.

Reviewer 4: ```suggestion
- **Reduce student frustration:** We give students a quick feedback loop. If a solution is optimal, let them continue. If it fails in a known way, tell them and let them improve it. This prevents students from becoming frustrated waiting for their perfectly-OK solutions from being approved reducing the perception of mentors being the source of slow approvals.
```
I think this reads a little easier.

Author: I've tweaked it slightly, but fundamentally what you wrote. Thanks :)
